server/lib/queries.js

```diff
@@ -126,6 +126,13 @@ module.exports = function(sequelize) {
         FROM "Donations" d
         WHERE d."GroupId" = :GroupId AND d.amount >= 0
         GROUP BY "UserId"
+      ), last_donation AS (
+        SELECT
+          max("UserId") as "UserId",
+          max("updatedAt") as "updatedAt"
+        FROM "Donations" d
+        WHERE d."GroupId" = :GroupId AND d.amount >= 0
+        GROUP BY "UserId"
```

Comment:
This will miss recurring donations. Recurring donations are only listed once in the `Donations` table with a `SubscriptionId`, that indicates that it's recurring. After that recurring donations are only stored in `Transactions` table. 

You could deal with that by left join `subscriptionid` and checking that `isActive` is true.